### PR TITLE
Updated osx_image to xcode8.2, close #334

### LINKED ISSUE #334: Update travis-ci to Xcode8.2
Travis has Xcode 8.2 GM now.
https://blog.travis-ci.com/2016-12-13-xcode-82-ga-is-here

Apparently 8.2 is supposed to be more reliable on CI. Let's see. 👀

### PR DESCRIPTION
**Pull request by heshamsalman**
## Changes in this pull request
- Updated `osx_image` in Travis to `xcode8.2`, as in: https://github.com/Instagram/IGListKit/issues/334
- Doesn't require tests, as it'll either pass with no changes to the codebase or break CI. 🤷🏽‍♂️

## Pull request checklist

- [x] All tests pass. Demo project builds and runs.
- [x] I added tests, an experiment, or detailed why my change isn't tested.
- [x] I have reviewed the [contributing guide](https://github.com/Instagram/IGListKit/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -36,6 +36,7 @@ env:
 
    - DESTINATION="OS=10.0,name=iPhone 6s" SDK="$IOS_SDK" SCHEME="$IOS_SCHEME" RUN_TESTS="YES" BUILD_EXAMPLE="YES" RUN_UI_TESTS="NO" POD_LINT="NO"
    - DESTINATION="OS=10.1,name=iPhone 6s" SDK="$IOS_SDK" SCHEME="$IOS_SCHEME" RUN_TESTS="YES" BUILD_EXAMPLE="NO"  RUN_UI_TESTS="YES" POD_LINT="NO"
+   - DESTINATION="OS=10.2,name=iPhone 6s" SDK="$IOS_SDK" SCHEME="$IOS_SCHEME" RUN_TESTS="YES" BUILD_EXAMPLE="NO"  RUN_UI_TESTS="YES" POD_LINT="NO"
```
Comment (jessesquires) [inline .travis.yml]: `RUN_UI_TESTS` should be `NO`

File: .travis.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ env:
    - TVOS_SCHEME="IGListKit-tvOS"
    - MACOS_SCHEME="IGListKit-macOS"
 
-   - IOS_SDK=iphonesimulator10.1
+   - IOS_SDK=iphonesimulator
```
Comment (jessesquires) [inline .travis.yml]: should be `iphonesimulator10.2`

File: .travis.yml [hunk 3]
```diff
@@ -14,7 +14,7 @@ env:
...
    - TVOS_SDK=appletvsimulator10.0
```
Comment (jessesquires) [inline .travis.yml]: should be `appletvsimulator10.1`

File: .travis.yml (hunk 1 above)
Comment (heshamsalman) [inline .travis.yml]: @jessesquires what's the rational for running UI/Unit tests on some targets but not others?

File: .travis.yml [hunk 4]
```diff
@@ -36,6 +36,7 @@ env:
 
    - DESTINATION="OS=10.0,name=iPhone 6s" SDK="$IOS_SDK" SCHEME="$IOS_SCHEME" RUN_TESTS="YES" BUILD_EXAMPLE="YES" RUN_UI_TESTS="NO" POD_LINT="NO"
    - DESTINATION="OS=10.1,name=iPhone 6s" SDK="$IOS_SDK" SCHEME="$IOS_SCHEME" RUN_TESTS="YES" BUILD_EXAMPLE="NO"  RUN_UI_TESTS="YES" POD_LINT="NO"
+   - DESTINATION="OS=10.2,name=iPhone 6s" SDK="$IOS_SDK" SCHEME="$IOS_SCHEME" RUN_TESTS="YES" BUILD_EXAMPLE="NO"  RUN_UI_TESTS="NO" POD_LINT="NO"
```
Comment (jessesquires) [inline .travis.yml]: ok, for some reason it looks like travis thinks `10.2, iPhone 6s` doesn't exist 😅 (it does)

let's try using `name=iPhone 7` for 10.2

File: .travis.yml [hunk 5]
```diff
@@ -37,6 +37,8 @@ env:
    - DESTINATION="OS=10.0,name=iPhone 6s" SDK="$IOS_SDK" SCHEME="$IOS_SCHEME" RUN_TESTS="YES" BUILD_EXAMPLE="YES" RUN_UI_TESTS="NO" POD_LINT="NO"
    - DESTINATION="OS=10.1,name=iPhone 6s" SDK="$IOS_SDK" SCHEME="$IOS_SCHEME" RUN_TESTS="YES" BUILD_EXAMPLE="NO"  RUN_UI_TESTS="YES" POD_LINT="NO"
 
+   - DESTINATION="OS=10.2,name=iPhone 7" SDK="$IOS_SDK" SCHEME="$IOS_SCHEME" RUN_TESTS="YES" BUILD_EXAMPLE="NO"  RUN_UI_TESTS="NO" POD_LINT="NO"
```
Comment (heshamsalman) [inline .travis.yml]: Ah, it's unable to find it due to a duplicate simulator for the 10.2 image. I'll use their suggested workaround until Travis fixes it on their end. 

https://github.com/travis-ci/travis-ci/issues/7031

File: .travis.yml (hunk 4 above)
Comment (heshamsalman) [inline .travis.yml]: It's unable to find the  due to a duplicate simulator for the 10.2 image. I'll use their suggested workaround until Travis fixes it on their end.

travis-ci/travis-ci#7031

**Comment by jessesquires** [comment]
thanks so much @heshamsalman !

it looks like the ipad simulators are *not* duplicated.

how about we switch to `iPad Air 2` for 10.2? then we won't need this workaround

**Comment by heshamsalman** [comment]
🤠 Just glad to contribute!

File: .travis.yml (hunk 1 above)
Comment (jessesquires) [inline .travis.yml]: @heshamsalman - oops, sorry. just saw this.

they are flaky, in general but specifically on travis too

**Pull request closed without merging**